internal/commands/pull/inputs.go

```diff
@@ -109,3 +117,64 @@ func (i *inputs) resolveRemoteApp(ui terminal.UI, clients cli.Clients) (realm.Ap
 
 	return app, nil
 }
+
+func (i *inputs) resolveClientTemplates(ui terminal.UI, realmClient realm.Client, groupID, appID string) (map[string]*zip.Reader, error) {
+	compatibleTemplates, err := realmClient.CompatibleTemplates(groupID, appID)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(compatibleTemplates) == 0 {
```

Comment:
not sure why this block moved, we should probably be keeping this _after_ the `if i.TemplateID != ""` block.

if they specified a template id but we got none back (likely due to the fact it was a `400` b/c the app wasn't created with a template), then we'll want to error, not proceed silently